reason: ppxlib migration

superherointj opened the pull request:
Highlight: `reason: ppxlib migration`

* refactoring
* add superherointj as maintainer
* ppxlib migration
* add downloadPage to meta
* removed maintainer volth by his request
* reason: add test hello

---

Maintainer `volth` answered e-mail stating:
```
Feel free from removing me as maintainer of Reason.
I am not using Reason, just evaluated it.
```
I have removed him.

vbgl commented on pkgs/development/compilers/reason/default.nix:
```diff
@@ -13,11 +13,27 @@ stdenv.mkDerivation rec {
+    menhirLib
+    merlin-extend
+    ppxlib
+  ];
```
Why is `ppxlib` propagated?

superherointj replied:
Forgive my ignorance. I was replacing ocaml-migrate-parsetree which was propagated. I incorrectly thought this should be too. I'll fix it.

superherointj replied:
Done.

vbgl replied:
BTW, it seems that `ppx_derivers` and `result` should be propagated.

superherointj replied:
I have created a demo (which likely should be added as a simple test) and it confirms the need of `ppx_derivers`. 
How do you know the need of `result`?

vbgl replied:
Indeed, this is surprising. When removing `ppx_derivers` from the propagated inputs of `reason`, the error log reads:

>  161 |  (requires compiler-libs.common result ppx_derivers)
                                               ^^^^^^^^^^^^
  Error: Library "ppx_derivers" not found.
  -> required by library "reason.ocaml-migrate-parsetree" in

this suggests that `result` is also required…

superherointj replied:
What if Result might be part of OCaml compiler already in newer version?

vbgl replied:
Well, don’t add `result` if you believe it’s not necessary.

superherointj replied:
Let's delay adding it until it breaks somewhere. I guess that will be the way to be sure of it.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 126066` at 9234c053 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>reason</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/reason/default.nix:40:3:
```
   |
40 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

siraben commented:
Result of `nixpkgs-review pr 126066` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>reason</li>
  </ul>
</details>

superherointj committed:
reason: refactoring

superherointj committed:
reason: add superherointj as maintainer

superherointj committed:
reason: ppxlib migration

Migrate from ocaml-migrate-parsetree to ppxlib fixing conflicts in downstream dependencies.

superherointj committed:
reason: add downloadPage to meta

superherointj committed:
reason: removed maintainer volth by his request

superherointj committed:
reason: add test hello
